### PR TITLE
Implement OAuth and add /login route

### PR DESCRIPTION
**Pull request by vgeorge**
Relates to #2, #3, #4.

Changelog:

- Added a migration for the user table. The schema is different than described at #4. I've kept only information provided by OSM, an internal id field and `isAdmin` for permission setting. I believe we don't need `password` and `email` field at the moment.
- Added `config` module for setting OAuth credentials or custom PostgreSQL instance via enviroment variables
- Added /login route, which implements OSM OAuth

Expected feedback:

- Are README instructions clear?
- Does tests pass?
- With OAuth credentials set, is it possible to login sucessfully by accessing http://localhost:3000/login?
- Is your user added to users table in PostgreSQL after login?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,20 +20,48 @@ Install Node.js dependencies:
 npm install
 ```
 
+### Configuration
+
+There are two ways of passing application options for PostgreSQL and OAuth:
+
+1) Using environment variables, full list available at [config/custom-environment-variables.json](config/custom-environment-variables.json)
+
+2) Adding file `config/local-development.json`. It will not be git-tracked and will override options for `development` environment.
```
Comment (batpad) [inline README.md]: It maybe useful to have a `config/local-development.json.example` or so checked in to the repository. It was not completely obvious how this file should look. Mine ended up being like this:

```
{
  "osmOAuth": {
    "requestTokenUrl": "https://master.apis.dev.openstreetmap.org/oauth/request_token",
    "accessTokenUrl": "https://master.apis.dev.openstreetmap.org/oauth/access_token",
    "authorizeUrl": "https://master.apis.dev.openstreetmap.org/oauth/authorize",
    "profileUrl": "https://master.apis.dev.openstreetmap.org/api/0.6/user/details",
    "clientId": "xxx",
    "clientSecret": "xxx"
  }
}
```

File: README.md [hunk 2]
```diff
@@ -18,20 +20,48 @@ Install Node.js dependencies:
...
+When finished, go back to the first terminal session and kill the database container with Ctrl+C.
+
 ### Testing
 
+Start test database:
+
+    npm run start-test-db
+
+There is no need to migrate the test database, it will be reset on every test run.
+
```
Comment (batpad) [inline README.md]: To remain consistent with README elsewhere, ask user to open a new terminal at this point to run the tests?

**Comment by batpad** [comment]
@vgeorge this looks really good and things seem to work pretty well!

> Are README instructions clear?

Have left comments for a couple of minor things. Overall I was able to get things up and running pretty easily.

One minor addition - it maybe good to add that one can set an environment variable for `PORT` to run on a port other than 3000. Not a huge deal, but this was useful for me as I had something else running on 3000.

> Does tests pass?

Yes.

> With OAuth credentials set, is it possible to login sucessfully by accessing http://localhost:3000/login?

I can access http://localhost:3000/login, I am redirected to the OSM dev login and then redirected back to the app where I get the JSON of some credentials.

What is not clear is how the client will use these credentials to authenticate subsequent requests. From looking at the `hapi/bell` usage [documentation](https://github.com/hapijs/bell/blob/6a3ededdcddc973e8b3adcb1ae1369f2b119cde8/API.md#usage), it does not in any way maintain the session state, and we will need another way to handle that.

Essentially, what would would be nice is another route that clearly demonstrates how the client would need to make requests to authenticate themselves -- something like `/profile` that just returns the OSM username of someone making an authenticated request, for example.

My preference would be for the backend to do the authentication with OSM, and then use something like https://github.com/dwyl/hapi-auth-jwt2 to generate a JWT token which will get handed down to the mobile application and that the client can then use for subsequent requests to the API to authenticate.

Whilst saying all this, the basic OAuth flow to OSM works, I think it would be okay if you want to handle this in a separate PR, @vgeorge - we will also need to figure out how the backend needs to redirect back to the mobile app with the credentials - am happy to talk through what we'd need to implement and test that.

> Is your user added to users table in PostgreSQL after login?

Yes. I wonder if some documentation on accessing the database maybe useful in the README or some developer docs - this is probably a bit confusing for someone not familiar with `docker`. What I landed up doing:

    docker ps # get the container id of the db container
    docker exec -it <container_id> bash
    psql -U observe-api-dev observe-api-dev

**Comment by vgeorge** [comment]
I've updated the instructions and merged. Discussed with @batpad that we will work on JWT in another PR.

**Pull request merged**